staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go

```diff
@@ -0,0 +1,203 @@
+		RESTClientGetter: restClientGetter,
+		PrintFlags:       genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+		IOStreams:        streams,
+	}
+}
+
+// AddFlags registers flags for a cli.
+func (flags *WhoAmIFlags) AddFlags(cmd *cobra.Command) {
+	flags.PrintFlags.AddFlags(cmd)
+}
+
+// ToOptions converts from CLI inputs to runtime inputs.
+func (flags *WhoAmIFlags) ToOptions(ctx context.Context, args []string) (*WhoAmIOptions, error) {
+	w := &WhoAmIOptions{
+		ctx:       ctx,
+		IOStreams: flags.IOStreams,
+	}
+
+	clientConfig, err := flags.RESTClientGetter.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	w.authClient, err = authenticationv1alpha1client.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	if !flags.PrintFlags.OutputFlagSpecified() {
+		w.resourcePrinterFunc = printTableSelfSubjectAccessReview
+	} else {
+		printer, err := flags.PrintFlags.ToPrinter()
+		if err != nil {
+			return nil, err
+		}
+		w.resourcePrinterFunc = printer.PrintObj
+	}
+
+	return w, nil
+}
+
+// WhoAmIOptions is the start of the data required to perform the operation. As new fields are added,
+// add them here instead of referencing the cmd.Flags()
+type WhoAmIOptions struct {
+	authClient authenticationv1alpha1client.AuthenticationV1alpha1Interface
+	ctx        context.Context
+
+	resourcePrinterFunc printers.ResourcePrinterFunc
+
+	genericclioptions.IOStreams
+}
+
+var (
+	whoAmILong = templates.LongDesc(`
+		Experimental: Check who you are and your attributes (groups, extra).
+
+        This command is helpful to get yourself aware of the current user attributes, 
+        especially when dynamic authentication, e.g., token webhook, auth proxy, or OIDC provider, 
+        is enabled in the Kubernetes cluster.
+	`)
+
+	whoAmIExample = templates.Examples(`
+		# Get your subject attributes.
+		kubectl alpha auth whoami
+		
+		# Get your subject attributes in JSON format.
+		kubectl alpha auth whoami -o json
+	`)
+)
+
+// NewCmdWhoAmI returns an initialized Command for 'auth whoami' sub command. Experimental.
+func NewCmdWhoAmI(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *cobra.Command {
+	flags := NewWhoAmIFlags(restClientGetter, streams)
+
+	cmd := &cobra.Command{
+		Use:                   "whoami",
+		DisableFlagsInUseLine: true,
+		Short:                 "Experimental: Check self subject attributes",
+		Long:                  whoAmILong,
+		Example:               whoAmIExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			o, err := flags.ToOptions(cmd.Context(), args)
+			cmdutil.CheckErr(err)
+			cmdutil.CheckErr(o.Run())
+		},
+	}
+
+	flags.AddFlags(cmd)
+	return cmd
+}
+
+// Run prints all user attributes.
+func (o WhoAmIOptions) Run() error {
+	sar := &authenticationv1alpha1.SelfSubjectReview{}
+	response, err := o.authClient.SelfSubjectReviews().Create(context.TODO(), sar, metav1.CreateOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
if the feature is not enabled on the server, the permission to run this will not be included in the basic-user role, and users are likely to get a forbidden error... it might be worth detecting forbidden errors and saying something like "The selfsubjectreviews API is not enabled in the cluster or you do not have permission to call it..." to clarify the possible reasons